### PR TITLE
(PC-33470) feat(tutorial): add AgeSelectionFork screen

### PR DESCRIPTION
**Pull request by lbeneston-pass**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-33470

## Flakiness

If I had to re-run tests in the CI due to flakiness, I add the incident [on Notion][2]

## Checklist

I have:

- [x] Made sure my feature is working on web.
- [x] Made sure my feature is working on mobile (depending on relevance : real or virtual devices)
- [x] Written **unit tests** native (and web when implementation is different) for my feature.
- [x] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion][1]
- [ ] I am aware of all the best practices and respected them.

## Screenshots

**delete** _if no UI change_

| Platform         | Mockup/Before | After |
| :--------------- | :-----------: | :---: |
| iOS              |               |   Capture d’écran 2024-12-23 à 12 46 31  |
| Android          |               |       |
| Phone - Chrome   |               |       |
| Desktop - Chrome |               |       |

[1]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a
[2]: https://www.notion.so/passcultureapp/cb45383351b44723a6f2d9e1481ad6bb?v=10fe47258701423985aa7d25bb04cfee&pvs=4

## Best Practices

<details>
  <summary>Click to expand</summary>
These rules apply to files that you make changes to.
If you can't respect one of these rules, be sure to explain why with a comment.
If you consider correcting the issue is too time consuming/complex: create a ticket. Link the ticket in the code.

- In the production code: remove type assertions with `as` (type assertions are removed at compile-time, there is no runtime checking associated with a type assertion. There won’t be an exception or `null` generated if the type assertion is wrong). In certain cases `as const` is acceptable (for example when defining readonly arrays/objects). Using `as` in tests is tolerable.
- Remove bypass type checking with `any` (when you want to accept anything because you will be blindly passing it through without interacting with it, you can use `unknown`). Using `any` in tests is tolerable.
- Remove non-null assertion operators (just like other type assertions, this doesn’t change the runtime behavior of your code, so it’s important to only use `!` when you know that the value can’t be `null` or `undefined`).
- Remove all `@ts-expect-error` and `@eslint-disable`.
- Remove all warnings, and errors that we are used to ignore (`yarn test:lint`, `yarn test:types`, `yarn start:web`...).
- Use `gap` (`ViewGap`) instead of `<Spacer.Column />`, `<Spacer.Row />` or `<Spacer.Flex />`.
- Don't add new "alias hooks" (hooks created to group other hooks together). When adding new logic, this hook will progressively become more complex and harder to maintain.
- Remove logic from components that should be dumb.

Test specific:

- Avoid mocking internal parts of our code. Ideally, mock only external calls.
- When you see a local variable that is over-written in every test, mock it.
- Prefer `user` to `fireEvent`.
- When mocking feature flags, use `setFeatureFlags`. If not possible, mention which one(s) you want to mock in a comment (example: `jest.spyOn(useFeatureFlagAPI, 'useFeatureFlag').mockReturnValue(true) // WIP_NEW_OFFER_TILE in renderPassPlaylist.tsx` )
- In component tests, replace `await act(async () => {})` and `await waitFor(/* ... */)` by `await screen.findBySomething()`.
- In hooks tests, use `act` by default and `waitFor` as a last resort.
- Make a snapshot test for pages and modals ONLY.
- Make a web specific snapshot when your web page/modal is specific to the web.
- Make an a11y test for web pages.

Advice:

- Use TDD
- Use Storybook
- Use pair programming/mobs
</details>

### CODE REVIEW

File: src/features/tutorial/components/AgeButton.stories.tsx [hunk 1]
```diff
@@ -120,11 +141,3 @@ const Template: VariantsStory<typeof AgeButton> = (args) => (
...
 export const AllVariants = Template.bind({})
 AllVariants.storyName = 'AgeButton'
-
-// export const DenseWithoutIcon = Template.bind({})
-// DenseWithoutIcon.args = {
-//   children: TextExample({}),
-//   dense: true,
-//   Icon: undefined,
-//   navigateTo: { screen: 'AgeSelection', params: { type: TutorialTypes.ONBOARDING } },
-// }
```
Comment (yleclercq-pass) [inline src/features/tutorial/components/AgeButton.stories.tsx:130]: un oublie ?

File: src/features/tutorial/pages/AgeSelectionFork.native.test.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  describe('onboarding', () => {
+    beforeEach(() => {
+      useRoute.mockReturnValueOnce({ params: { type: TutorialTypes.ONBOARDING } })
+    })
+
+    it('should navigate to Home page when pressing "J’ai 14 ans ou moins"', async () => {
+      renderAgeSelectionFork({ type: TutorialTypes.ONBOARDING })
+
+      const button = screen.getByLabelText('J’ai 14 ans ou moins')
+      fireEvent.press(button)
```
Comment (yleclercq-pass) [inline src/features/tutorial/pages/AgeSelectionFork.native.test.tsx:30]: Je crois qu'on essaye d'utiliser userEvent désormais tu as quelques exemple dans le code comment l'utiliser. Je te laisse voir pour le changer sur tous les tests de ce fichier avec `fireEvent `

File: src/features/tutorial/pages/AgeSelectionFork.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      endButtonTitle: ' ou moins',
+      navigateTo: navigateToHomeConfig,
+      onBeforeNavigate: () => showNonEligibleModal(NonEligible.UNDER_15, type),
+    },
+    {
+      startButtonTitle: 'J’ai entre ',
+      age: '15 et 18 ans',
+      endButtonTitle: '',
+      navigateTo: { screen: 'EligibleUserAgeSelection' },
+      onBeforeNavigate: undefined,
```
Comment (yleclercq-pass) [inline src/features/tutorial/pages/AgeSelectionFork.tsx:42]: je ne pense pas que tu aies besoin de renseigner la propriété `onBeforeNavigate,` idem pour celle d'en dessous. 
après si le typage ne le permet pas on peut l'adapter : `onBeforeNavigate?: (() => void) | null;`

File: src/features/tutorial/pages/AgeSelectionFork.native.test.tsx (hunk 2 above)
Comment (lbeneston-pass) [reply src/features/tutorial/pages/AgeSelectionFork.native.test.tsx:30]: J'ai essayé et ça ne fonctionne pas 🤷

File: src/features/tutorial/components/AgeButton.stories.tsx (hunk 1 above)
Comment (lbeneston-pass) [reply src/features/tutorial/components/AgeButton.stories.tsx:130]: Oui d'ou la suppression ! C'était commenté

File: src/features/tutorial/pages/AgeSelectionFork.native.test.tsx (hunk 2 above)
Comment (yleclercq-pass) [reply src/features/tutorial/pages/AgeSelectionFork.native.test.tsx:30]: tu as tenté de rajouter ? car cela va souvent avec
`  jest.useFakeTimers()`

**Review by yleclercq-pass — APPROVED** [review]

**Pull request merged**